x/starname/keeper/querier.go

```diff
@@ -314,3 +315,55 @@ func queryBrokerDomains(ctx sdk.Context, keeper *Keeper, broker sdk.AccAddress,
 	}
 	return &types.QueryBrokerDomainsResponse{Domains: domains, Page: page}, nil
 }
+
+//Yield return an estimation of the delegators annualized yield based on the last 100k blocks
+func (q grpcQuerier) Yield(ctx context.Context, _ *types.QueryYieldRequest) (*types.QueryYieldResponse, error) {
+	var response types.QueryYieldResponse
+
+	apy, err := calculateYield(sdk.UnwrapSDKContext(ctx), q.keeper)
+	if err != nil {
+		return nil, err
+	}
+	response.Apy = apy
+	return &response, err
+}
+
+func calculateYield(ctx sdk.Context, keeper *Keeper) (sdk.Dec, error) {
+	// TODO: dont rely on an hardcoded estimate
+	const EstimatedBlockTime = 4
+
+	totalFees, numBlocks := keeper.GetBlockFeesSum(ctx)
+
+	// Compute the reward pool from the collected fees for the last numBlocks blocks
+	communityTax := keeper.DistributionKeeper.GetCommunityTax(ctx)
+	// communityReward = fees * communityTax
+	// rewardPool = fees - communityReward
+	// => reward = fees * (1 - communityTax)
+	rewardPool := sdk.NewDecCoinsFromCoins(totalFees...).
+		MulDec(sdk.OneDec().Sub(communityTax))
+
+	totalDelegatedTokens := keeper.StakingKeeper.GetLastTotalPower(ctx) // in iov
+
+	// Voting power is returned in tokens while fees are in a sub-unit (iov vs uiov)
+	multiplier := 1e6
+	totalDelegatedTokens = totalDelegatedTokens.Mul(sdk.NewInt(int64(multiplier))) // in uiov
+
+	// Compute yield for numBlocks blocks
+	yieldForPeriod := rewardPool.QuoDec(sdk.NewDecFromInt(totalDelegatedTokens))
+
+	var apy sdk.Dec
+	if len(yieldForPeriod) == 0 {
+		apy = sdk.ZeroDec()
+	} else {
+		const SecondsPerYear = 365 * 24 * 3600
+		numBlocksPerYear := SecondsPerYear / EstimatedBlockTime
+
+		// TODO: manage multiple tokens for fees
+		apy = yieldForPeriod.
```

Comment:
We assume that 100k blocks equals 1 week and that there are 52 weeks in a year, so simply multiply `yieldForPeriod` by 52.  This is just an estimate on yield, so we can take liberties.